internal/driver/glfw/menu_bar_test.go

```diff
@@ -428,18 +428,20 @@ func TestMenuBar(t *testing.T) {
 				test.TapCanvas(c, fileMenuPos) // activate menu
 				require.Equal(t, menuBar.Items[0], c.Focused())
 
+				var captured image.Image
 				runOnMain(func() {
-					if test.AssertImageMatches(t, "menu_bar_active_file.png", c.Capture()) {
-						lastAction = ""
-						for _, key := range tt.keys {
-							c.Focused().TypedKey(&fyne.KeyEvent{
-								Name: key,
-							})
-						}
-						test.AssertRendersToMarkup(t, "menu_bar_kbdctrl_"+name+".xml", c)
-						assert.Equal(t, tt.wantAction, lastAction, "last action should match expected")
-					}
+					captured = c.Capture()
```

Comment:
Capture on safeCanvas is already safe isn't it?

Reply:
Oh, I see this is a test window. Maybe that is where some races are coming from? Mixing the test and GLFW windows?

Reply:
Yeah, exactly. I got strange errors when I changed to the safe window. Mixing it in the same package does make the app run in the background and race on canvas internals